### PR TITLE
Add licenses target to report them

### PR DESCRIPTION
**Pull request by lindig**
Add Makefile target "licenses" to report the licenses of all toolstack
packages.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

### CODE REVIEW

**Comment by psafont** [comment]
The command is a bit awkward to run since it implicitly needs an active switch with the xs-opam repo only, might be worth at least documenting it in the Makefile.

It would be interesting to convert this into some sort of check in the CI on the long run so we take care of the licenses in the metadata. Right now most xapi-project packages are missing the information.

**Comment by lindig** [comment]
I now see what you mean: this does not work on the data in repo but on the data that has been installed from the repo, which it not intuitive. Would it be better to put this in a script under `tools/`?

**Comment by psafont** [comment]
> I now see what you mean: this does not work on the data in the repo but on the data that has been installed from the repo, which it not intuitive. Would it be better to put this in a script under `tools/`?

I'd be happy with that plus the comment explaining how to run it successfully.

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,11 @@ $(NAME).tar.gz:
 	mv ocaml packages
 	mv upstream-extra packages
 
+licenses:
+	opam list --required-by "xs-toolstack" --recursive --short \
+	  | xargs opam info -f name:,license: | tr -d '"' \
+	  | awk '/^name:/ { name = $$2 } /^license:/ { license = $$2; printf("%s: %s\n",name, license); }'
+
```
Comment (psafont) [inline Makefile:31]: Did you intend to keep the command in both files?

Comment (lindig) [reply Makefile]: Sorry - missed adding the updated Makefile

**Pull request merged**